src/main/java/net/mcreator/workspace/misc/WorkspaceInfo.java

```diff
@@ -148,32 +148,48 @@ public Map<String, List<MItemBlock>> getCreativeTabMap() {
 					for (TabEntry tabEntry : tabElement.getCreativeTabs()) {
 						String tab = tabEntry.getUnmappedValue();
 
-						// If tab does not have custom order, add items to the end of the list
-						if (workspace.getCreativeTabsOrder().get(tab) == null)
-							tabMap.computeIfAbsent(tab, key -> new ArrayList<>()).addAll(tabItems);
+						// If tab does not have custom order, add items to the end of appropriate list
+						if (workspace.getCreativeTabsOrder().get(tab) == null) {
+							(tab.startsWith("CUSTOM:") ? customTabsWithoutOrder : tabMap).computeIfAbsent(tab,
+									key -> new ArrayList<>()).addAll(tabItems);
+						}
 					}
 				}
 			}
 		}
 
-		// Last, we add items to tabs with custom order
-		for (String tab : tabMap.keySet()) {
-			ArrayList<String> tabOrder = workspace.getCreativeTabsOrder().get(tab);
-			if (tabOrder != null) {
-				for (String element : tabOrder) {
-					ModElement me = workspace.getModElementByName(element);
-					if (me != null && me.getGeneratableElement() instanceof ITabContainedElement tabElement) {
-						List<MCItem> tabItems = tabElement.getCreativeTabItems();
-						if (tabItems != null && !tabItems.isEmpty()) {
-							tabMap.computeIfAbsent(tab, key -> new ArrayList<>())
-									.addAll(tabItems.stream().map(e -> new MItemBlock(workspace, e.getName()))
-											.toList());
-						}
+		// Next, we add items to tabs with custom order
+		for (Map.Entry<String, ArrayList<String>> entry : workspace.getCreativeTabsOrder().entrySet()) {
+			String tab = entry.getKey();
+			if (tab.startsWith("CUSTOM:")) {
+				ModElement tabME = workspace.getModElementByName(tab.replace("CUSTOM:", ""));
+				if (tabME == null || tabME.getType() != ModElementType.TAB)
+					continue; // Might be a stale entry we didn't remove previously (#5174)
+			}
+			for (String element : entry.getValue()) {
+				ModElement me = workspace.getModElementByName(element);
+				if (me != null && me.getGeneratableElement() instanceof ITabContainedElement tabElement) {
+					List<MCItem> tabItems = tabElement.getCreativeTabItems();
+					if (tabItems != null && !tabItems.isEmpty()) {
+						tabMap.computeIfAbsent(tab, key -> new ArrayList<>())
+								.addAll(tabItems.stream().map(e -> new MItemBlock(workspace, e.getName()))
+										.toList());
 					}
 				}
 			}
 		}
 
+		// Last, we move custom tabs to the end of the main list

```

Comment:
Why is this part needed? Does it matter for the templates that custom entries are at the end of the list?

Reply:
Since I really prefer to keep communication in one place:

![image](https://github.com/user-attachments/assets/e40860fd-fa6d-4095-b725-a31939762e6e)

Reply:
![image](https://github.com/user-attachments/assets/05d0ba0c-8cef-44e3-a0e2-beb6089bce1a)